pkg/controller/build/helpers.go

```diff
@@ -184,5 +117,156 @@ func validateSecret(secretGetter func(string) (*corev1.Secret, error), mosc *mcf
+	if !oldState.HasBuildConditions() && curState.HasBuildConditions() && curState.IsInInitialState() {
+		return true, "in initial state"
+	}
+
+	oldTransientState := oldState.GetTransientState()
+	curTransientState := curState.GetTransientState()
+
+	// From initial state -> pending or building.
+	if oldState.IsInInitialState() && curState.IsInTransientState() {
+		return true, fmt.Sprintf("transitioned from initial state -> transient state (%s)", curTransientState)
+	}
+
+	// From pending -> building.
+	if oldState.IsInTransientState() && curState.IsInTransientState() && oldTransientState != curTransientState {
+		return true, fmt.Sprintf("transitioned from transient state (%s) -> transient state (%s)", oldTransientState, curTransientState)
+	}
+
+	oldTerminalState := oldState.GetTerminalState()
+	curTerminalState := curState.GetTerminalState()
+
+	// From building -> {success, failure, interrupted}
+	if oldState.IsInTransientState() && curState.IsInTerminalState() {
+		return true, fmt.Sprintf("transitioned from transient state (%s) -> terminal state (%s)", oldTransientState, curTerminalState)
+	}
+
+	// From initial state -> {success, failure, interrupted}
+	// It's rare that this could occur, but better to be explicit that it can occur.
+	if oldState.IsInInitialState() && curState.IsInTerminalState() {
+		return true, fmt.Sprintf("transitioned from initial state -> terminal state (%s)", curTerminalState)
+	}
+
+	// Once a build enters a terminal state, it cannot transition to any other
+	// terminal, transient, or initial state.
+
+	// From {success, failure, interrupted} -> {success, failure, interrupted}
+	if oldState.IsInTerminalState() && curState.IsInTerminalState() {
+		return false, fmt.Sprintf("transitioned from terminal state (%s) -> terminal state (%s)", oldTerminalState, curTerminalState)
+	}
+
+	// From {success, failure, interrupted} -> {pending, running}
+	if oldState.IsInTerminalState() && curState.IsInTransientState() {
+		return false, fmt.Sprintf("transitioned from terminal state (%s) -> transient state (%s)", oldTerminalState, curTransientState)
+	}
+
+	// From {sucecss, failure, interrupted} -> initial state
+	if oldState.IsInTerminalState() && curState.IsInInitialState() {
+		return false, fmt.Sprintf("transitioned from terminal state (%s) -> initial state", oldTerminalState)
+	}
+
+	// Everything else
+	return false, ""
+}
+
+// Converts a list of MachineOSConfigs into a list of their names.
+func getMachineOSConfigNames(moscList []*mcfgv1alpha1.MachineOSConfig) []string {
+	out := []string{}
+
+	for _, mosc := range moscList {
+		out = append(out, mosc.Name)
+	}
+
+	return out
+}
+
+// Converts a list of MachineOSBuilds into a list of their names.
+func getMachineOSBuildNames(mosbList []*mcfgv1alpha1.MachineOSBuild) []string {
+	out := []string{}
+
+	for _, mosc := range mosbList {
+		out = append(out, mosc.Name)
+	}
+
+	return out
+}
+
+// Determines if a MachineOSBuild is current for a given MachineOSConfig soley
```

Comment:
Replace "soley" with "solely" to get `make verify` pass

Reply:
Fixed.